test/widgets/chat/chat_page_test.dart

```diff
@@ -0,0 +1,176 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:motis_mitfahr_app/account/models/profile.dart';
+import 'package:motis_mitfahr_app/util/chat/chat_bubble.dart';
+import 'package:motis_mitfahr_app/util/chat/message_bar.dart';
+import 'package:motis_mitfahr_app/util/chat/pages/chat_page.dart';
+import 'package:motis_mitfahr_app/util/profiles/profile_widget.dart';
+import 'package:motis_mitfahr_app/util/supabase.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+import '../../util/factories/message_factory.dart';
+import '../../util/factories/profile_factory.dart';
+import '../../util/mock_server.dart';
+import '../../util/pump_material.dart';
+import '../../util/request_processor.dart';
+import '../../util/request_processor.mocks.dart';
+
+void main() {
+  final MockRequestProcessor processor = MockRequestProcessor();
+  final Profile profile = ProfileFactory().generateFake(id: 1);
+  const int chatId = 1;
+
+  setUpAll(() {
+    MockServer.setProcessor(processor);
+  });
+
+  testWidgets('Page has stream subscription', (WidgetTester tester) async {
+    await pumpMaterial(tester, ChatPage(profile: profile, chatId: chatId));
+    verifyRequest(
+      processor,
+      urlMatcher: equals('/rest/v1/messages?select=%2A&chat_id=eq.$chatId&order=created_at.desc.nullslast'),
+    ).called(1);
+    final List<RealtimeChannel> subscription = SupabaseManager.supabaseClient.getChannels();
+    expect(subscription.length, 1);
+    expect(subscription[0].topic, 'realtime:public:messages:1');
+  });
+  testWidgets('shows ProfileWidget of given Profile', (WidgetTester tester) async {
+    await pumpMaterial(tester, ChatPage(profile: profile, chatId: chatId));
+    final Finder profileWidgetFinder = find.byType(ProfileWidget);
+    expect(profileWidgetFinder, findsOneWidget);
+    final ProfileWidget profileWidget = tester.widget(profileWidgetFinder);
+    expect(profileWidget.profile, profile);
+  });
+
+  group('Messagebar', () {
+    setUp(() => whenRequest(processor).thenReturnJson([]));
```

Comment:
can you use a few more line breaks after setUp and between the testWidgets themselves :) Would be a lot easiert to read

Reply:
did that